pkg/fanal/analyzer/language/golang/mod/mod.go

```diff
@@ -341,66 +337,60 @@ func mergeGoSum(gomod, gosum *types.Application) {
 	gomod.Packages = lo.Values(uniq)
 }
 
-func findLicense(fsys fs.FS, vendorDirFound bool, dir string, classifierConfidenceLevel float64) ([]string, error) {
-	var license *types.LicenseFile
-
-	open := func(fsys fs.FS, path string) (fs.File, error) {
-		if vendorDirFound {
-			return fsys.Open(path)
-		}
-		return os.Open(path)
+func findLicense(searchDirs []searchDir, pkg types.Package, licenses map[string][]string, classifierConfidenceLevel float64) ([]string, error) {
+	if licenses[pkg.ID] != nil {
+		return licenses[pkg.ID], nil
 	}
 
-	walkDirFunc := func(path string, d fs.DirEntry, err error) error {
+	var license *types.LicenseFile
+	for _, searchDir := range searchDirs {
+		// e.g. $GOPATH/pkg/mod => $GOPATH/pkg/mod/github.com/aquasecurity/go-dep-parser@v0.1.0
```

Comment:
This comment might be confusing for the vendor dir.
So let's move on to the `Resolve` function in the interface:
https://github.com/aquasecurity/trivy/blob/ec358e42554041eb447eeb960647a415c15d676b/pkg/fanal/analyzer/language/golang/mod/mod.go#L412
Or just add an example for vendor dir.

Reply:
I added more comments to each `Resolve` method.
https://github.com/aquasecurity/trivy/pull/8843/commits/e8b55ebb6df0e712b180dab921fd31af075b701f